### ISSUE TITLE
OpenTracing integration test fails with Camel 3.5.0-SNAPSHOT

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#1493]
Until now we seem to have avoided OpenTracing API incompatibilities between 0.31.0 (used by Quarkus) & 0.33.0 (used by Camel). The Camel `OpenTracingTracer` now invokes a method that is not present in 0.31.0, hence we see:

```
[INFO] Running org.apache.camel.quarkus.component.opentracing.it.OpenTracingTest
2020-07-22 09:34:37,146 INFO  [org.apa.cam.qua.cor.CamelBootstrapRecorder] (main) bootstrap runtime: org.apache.camel.quarkus.main.CamelMainRuntime
2020-07-22 09:34:37,184 INFO  [org.apa.cam.mai.BaseMainSupport] (main) Auto-configuration summary:
2020-07-22 09:34:37,185 INFO  [org.apa.cam.mai.BaseMainSupport] (main) 	camel.context.name=quarkus-camel-example
2020-07-22 09:34:37,205 INFO  [org.apa.cam.qua.mai.CamelMainRoutesCollector] (main) Loading additional Camel XML route templates from: classpath:camel-template/*.xml
2020-07-22 09:34:37,226 ERROR [io.qua.application] (main) Failed to start application: java.lang.NoSuchMethodError: 'boolean io.opentracing.util.GlobalTracer.registerIfAbsent(io.opentracing.Tracer)'
	at org.apache.camel.opentracing.OpenTracingTracer.registerGlobalTracer(OpenTracingTracer.java:234)
	at org.apache.camel.opentracing.OpenTracingTracer.doInit(OpenTracingTracer.java:228)
	at org.apache.camel.support.service.BaseService.init(BaseService.java:83)
	at org.apache.camel.support.service.ServiceHelper.initService(ServiceHelper.java:55)
	at org.apache.camel.impl.engine.AbstractCamelContext.internalAddService(AbstractCamelContext.java:1485)
	at org.apache.camel.impl.engine.AbstractCamelContext.addService(AbstractCamelContext.java:1403)
	at org.apache.camel.opentracing.OpenTracingTracer.init(OpenTracingTracer.java:126)
	at org.apache.camel.opentracing.OpenTracingTracer.createRoutePolicy(OpenTracingTracer.java:114)
```

AFAIK - Quarkus will not move to OpenTracing 0.33.x until MicroProfile OpenTracing is upgraded to 2.0.

**Comment by davsclaus** [comment]
They are on 0.33 on their master branch
https://github.com/eclipse/microprofile-opentracing/blob/master/pom.xml#L38

**Comment by davsclaus** [comment]
Maybe in the mean time we can make OpenTracingTracer in camel use the older api if its possible

**Comment by davsclaus** [comment]
Quarkus 1.7 seems to be going with MP 1.3.3 that uses the older 0.31.0 version. 

So we may look at making camel-opentracing use the older API if possible and then upgrade to 0.33+ later.

**Comment by davsclaus** [comment]
This is only relevant when upgrading to Camel 3.5, and for quarkus 1.7 we can stay on camel 3.4.2 release.

**Comment by davsclaus** [comment]
Okay we have dropped back to 0.31.0 version
https://issues.apache.org/jira/browse/CAMEL-15354

**Comment by jamesnetherton** [comment]
I verified this is fixed with the latest Camel 3.5.0 SNAPSHOT.
